signer/signer.go

```diff
@@ -0,0 +1,483 @@
+package signer
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha1"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"errors"
+	"fmt"
+	"math/big"
+	"strconv"
+	"strings"
+	"time"
+
+	ct "github.com/google/certificate-transparency-go"
+	cttls "github.com/google/certificate-transparency-go/tls"
+	ctx509 "github.com/google/certificate-transparency-go/x509"
+	"github.com/jmhodges/clock"
+	"github.com/letsencrypt/boulder/cmd"
+	zlintx509 "github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/v2"
+	"github.com/zmap/zlint/v2/lint"
+)
+
+// IssuanceRequest describes a certificate issuance request
+type IssuanceRequest struct {
+	PublicKey crypto.PublicKey
+
+	Serial []byte
+
+	NotBefore time.Time
+	NotAfter  time.Time
+
+	CommonName string
+	DNSNames   []string
```

Comment:
For the sake of future compatibility with ACME-IP, should this just directly be SANs instead of DNSNames?

Reply:
I think replacing with, say, `[]core.AcmeIdentifier` starts bleeding a bit too much of the ACME API into the signer. When we add IPs I think it'd be reasonable to just add a `IPAddresses []net.ip` field to the struct.

Reply:
SGTM